### DISCUSSION TITLE
[tiny, low prio] Syntax batch operations: lock-unlock

### DISCUSSION THREAD

**Discussion by mietcls** [ugent-library/biblio-backend#1513] [Q&A] [answered]
Is there a reason to derive from the batch operations syntax for locking and unlocking?

Only locking and unlocking have a deriving structure, which makes it hard to remember what the structure is (or create templates).
<img width="581" alt="Screenshot 2024-05-08 at 14 07 36" src="https://github.com/ugent-library/biblio-backoffice/assets/1572885/3ba2deb6-d918-4e63-8a9c-719bcf6409f0">

Suggestions, following our existing syntax:
`8701504,locking.set,locked`
`8701504,locking.set,unlocked`

**Comment by nicolasfranck** [comment]
`true|false` could also be an option

**Comment by nics** [comment, marked as answer]
@mietcls the only syntax is operation name + zero or more arguments. The one's we have now with a dot operate on a property. But there are also object level operations like lock, unlock, delete, ... where it feels artificial to say `deleted.set, true` instead of simply `delete`.
The dots have no meaning, we can rename `project.add` to `add_project` if that feels more natural?

**Reply by mietcls** [reply]
Thank you for your feedback! That indeed feels more natural and makes it easy to remember / structure.
Not urgent, going to archive this discussion until there is a need to rework/extend or batch operations in a significant way in the future.
